installers/msi-language/ActiveState/Tracker.cs

```diff
@@ -58,12 +58,20 @@ public async Task TrackS3Event(Session session, string sessionID, string categor
             session.Log(string.Format("Downloading S3 pixel from URL: {0}", pixelURL));
             try
             {
-                var client = new TimeoutWebClient();
-                // never attempt to send pixel for more than 15 seconds, as it blocks the entire MSI
-                client.Timeout = 15 * 1000;
-                await client.DownloadStringTaskAsync(pixelURL);
+                // retry up to 3 times to download the S3 pixel
+                RetryHelper.RetryOnException(session, 3, TimeSpan.FromSeconds(1), () =>
+                {
+                    Task.Run(async () =>
+                    {
+                        var client = new TimeoutWebClient();
+                        // tr tp complete an s3 tracking event in seven seconds or less.
+                        client.Timeout = 7 * 1000;
+                        var res = await client.DownloadStringTaskAsync(pixelURL);
```

Comment:
Could we just use `client.DownloadString` rather than this async method if we are just going to wait for it anyways?

Reply:
That is a good point.